src/server/controllers/getPhotoMemory.js

```diff
@@ -0,0 +1,23 @@
+const aws = require('aws-sdk');
+require('env2')('./config.env');
+
+const S3_BUCKET_NAME = process.env.S3_BUCKET_NAME;
+
+module.exports = (req, res) => {
+  aws.config.update({ accessKeyId: process.env.AWSAccessKeyId, secretAccessKey: process.env.AWSSecretKey, signatureVersion: 'v4', region: 'eu-west-2' });
+  const s3 = new aws.S3();
+
+  const s3Params = {
+    Bucket: S3_BUCKET_NAME,
+    Key: req.query.memoryUrl,
```

Comment:
Okay, there are 2 things you are missing here.

Firstly, you need to check if this memory belongs to the user!

I would

  1. Only keep the memoryId on the client (not the s3 key, helpfully called memoryUrl)
  2. In the endpoint, use the id from the session (this means that the user has been authenticated), and the memoryId, to find a the s3key for the memory.
  3. Make sure you check the memory belongs to the authenticated user!